assets/js/phoenix_live_view/dom.js

```diff
@@ -360,7 +360,11 @@ let DOM = {
     let sourceAttrs = source.attributes
     for(let i = sourceAttrs.length - 1; i >= 0; i--){
       let name = sourceAttrs[i].name
-      if(exclude.indexOf(name) < 0){ target.setAttribute(name, source.getAttribute(name)) }
+      if(exclude.indexOf(name) < 0){
+        if(target.getAttribute(name) !== source.getAttribute(name)){
+          target.setAttribute(name, source.getAttribute(name))
+        }
+      }
```

Comment:
Regarding performance, the additional check has a small cost, but the existing code could be optimized, as `exclude.indexOf` is being called within a loop of the `sourceAttrs. 

Converting `exclude` to a `Set` for those checks would [in most cases](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#performance) avoid  nested iteration under the hood.

It's existing code, but a small change to make, and a good pattern to follow.